uwblueprint/community-fridge-kw
# feat: Delete User Backend Changes

## Brief description. What is this change? 

### [Delete User Backend Changes](https://www.notion.so/uwblueprintexecs/Delete-user-changes-193fe48e0f4c4cf8b9080fcaef593ed0)



## Implementation description. How did you make this change?
* When a user is deleted, we also have to delete / update different tables depending on whether the user is a donor or volunteer.
* When a volunteer is deleted from the database, the food rescues and the general checkins that they are associated with should have the `volunteerId` field cleared.
* When a donor is deleted from the database, the schedules that they have created are also deleted.
* For this pr, `deleteUserById` function is changed accordingly for the use cases mentioned above

TODOOOO: Write and update tests


## Steps to test
1. Create a donor account with an actual email and verify it. Lmk if you want to delete one from Firebase.
2. Create multiple different schedules as a donor
3. Create checkins using postman (since we dont have FE ready for it yet) with a body similar to the following:
```
{
    "startDate": "2022-03-14T09:00:00.000Z",
    "endDate": "2022-03-28T10:00:00.000Z",
    "notes": "my notes"
}
```
4. Create a volunteer account with an actual email and verify it.
5. Update some created schedules (`localhost:5000/scheduling/:id`)and checkins (`localhost:5000/checkin/:id`) on postman with a following payload (You can check the volunteerId from pgadmin by looking at the data in volunteers table or make a get call on `localhost:5000/volunteers` and see the id of the volunteer you created). You can do the same thing to find the ids of the checkins and schedules created
```
{
  "volunteerId" : "2"
}
```
6. To delete users, you will need to remove the `isAuthorizedByRole(new Set([Role.ADMIN])),` in the delete function in `userRoutes.ts` file so that you can use the user routes without actually being an admin
6. Delete volunteer using postman `/users?userId=<userId>` and verify the volunteer_id of the schedules and checkins you edited is now null and volunteer is deleted from both users and volunteers table. Again you can find the `userId` of the volunteer by making the get call on `localhost:5000/volunteers` and see the id of the volunteer you created. 
7. Delete donor using postman `/users?userId=<userId>` and verify the schedules associated with the donor is deleted and donor is deleted from both users and donors table




## Checklist
